src/DependencyInjection/Compiler/MapFragmentsToGlobalsPass.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2017 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\DependencyInjection\Compiler;
+
+use Contao\ContentProxy;
+use Contao\CoreBundle\EventListener\GlobalsMapListener;
+use Contao\CoreBundle\Fragment\Reference\ContentElementReference;
+use Contao\CoreBundle\Fragment\Reference\FrontendModuleReference;
+use Contao\ModuleProxy;
+use Symfony\Component\Config\Definition\Exception\InvalidConfigurationException;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\Compiler\PriorityTaggedServiceTrait;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+
+class MapFragmentsToGlobalsPass implements CompilerPassInterface
+{
+    use PriorityTaggedServiceTrait;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container): void
+    {
+        $elements = $this->getFragmentTags($container, ContentElementReference::TAG_NAME);
+        $elements = $this->createGlobalsMapForTag($elements, 'TL_CTE', ContentProxy::class);
+
+        $modules = $this->getFragmentTags($container, FrontendModuleReference::TAG_NAME);
+        $modules = $this->createGlobalsMapForTag($modules, 'FE_MOD', ModuleProxy::class);
+
+        $listener = new Definition(GlobalsMapListener::class, [array_merge_recursive($elements, $modules)]);
+        $listener->addTag('contao.hook', ['hook' => 'initializeSystem', 'priority' => 255]);
```

Comment:
Nice one :) A unit test should cover that this pass is registered before @dmolineus new hook register pass :)

Reply:
Added in 0e24990238a859115532b603e48f363859cdb614